alfalfa_worker/dispatcher.py

```diff
@@ -0,0 +1,139 @@
+
+
+class Dispatcher(DispatcherLoggerMixin, AlfalfaConnectionsBase):
+    """Class to pop data off a queue and determine to where the work needs
+    to go. Alfalfa works off a single queue and the work is identified
+    based on the payload that is in the queue (model_name).
+
+    This class is currently designed to do the work on the worker that it is
+    attached to. That is, the Dispatcher is not designed to pop work off the
+    queue and submit the job to another worker with the requisite files.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.logger.info(f"Job queue url is {self.sqs_queue}")
+
+        # create classes for the workers that this dispatcher supports
+        self.worker_openstudio_class = WorkerOpenStudio()
+        self.worker_fmu_class = WorkerFmu()
+
+    def determine_worker_class(self, test_str):
+        """This is the only place where we should decide
+        if the work is an OSW or an FMU
+        """
+        worker_class = None
+        test_str = test_str.lower().replace('.', '')
+        if test_str in ['osw', 'osm', 'zip']:
+            worker_class = self.worker_openstudio_class
+        elif test_str == 'fmu':
+            worker_class = self.worker_fmu_class
+
+        if worker_class is None:
+            raise Exception(f"Unable to determine worker class of string {test_str}")
+
+        return worker_class
+
+    def process_message(self, message):
+        """Process a single message from Queue.  Depending on operation requested,
+        will call one of:
+
+            - step_sim
+            - add_site
+            - run_sim
+        """
+        try:
+            message_body = json.loads(message.body)
+            self.logger.info(message_body)
+            message.delete()
+            op = message_body.get('op')
+            if op == 'InvokeAction':
+                action = message_body.get('action')
+                if action in ['init', 'addSite']:
+                    # get the model type from the body to determine which class will
+                    # dispatch the work
+                    model_name = Path(message_body.get('model_name'))
+
+                    worker_class = self.determine_worker_class(model_name.suffix)
+                    self.logger.info(f"Dispatching {action} with model_type {model_name.suffix} to worker {worker_class}")
+                    if action == 'init':
+                        # For testing purposes, just return the
+                        # worker class that was assigned
+                        return worker_class
+                    elif action == 'addSite':
+                        worker_class.add_site(message_body)
+                elif action in ['runSite', 'runSim']:
+                    # get the site ID out of the message
+                    site_id = message_body.get('id')
+                    site_rec = self.mongo_db_recs.find_one({"_id": site_id})
```

Comment:
sorry, had to pull this from mongo. This is also in each of the run* files. This method should get abstracted regardless, perhaps in a "Site" class (similar to an ORM), and then we can easily replace with Postgres or the like.

Reply:
#233 will impact this as well i think?

Reply:
yes it will. I think we need to make a `Site` class that knows how to talk to the database/mongo. This is starting to look like an ORM/ODM need...